### DISCUSSION TITLE
Pass all required checks

### DISCUSSION THREAD

**Discussion by asbjornu** [Mergifyio/mergify#2184] [Q&A]
I understand that something like `check-success=*` can't work, since GitHub checks are reported asynchronously. Some sort of hard coding of the required checks are needed.

Which is why I have to ask: Can't [required checks](https://docs.github.com/en/github/administering-a-repository/managing-a-branch-protection-rule) be used for this precise purpose? Required checks are hard coded into the branch settings of the repository, so `check-success-required=true` or something similar should work, no?

Having to hard code the required checks both in the repository settings and into `.mergify.yml` at least feels unnecessary.

**Comment by jd** [comment]
You're right @asbjornu.
You can use the branch protection settings with whatever settings you want, Mergify will honor them automatically.
https://docs.mergify.io/actions/merge.html#branch-protection-settings

**Reply by asbjornu** [reply]
Indeed, that works like a charm! Thanks!
